tests/unit/lib/package/test_artifact_exporter.py

```diff
@@ -1636,6 +1661,46 @@ def test_make_zip_windows(self, mock_system):
                 os.remove(zipfile_name)
             test_file_creator.remove_all()
 
+    def test_make_zip_lambda_resources(self):
+
+        test_file_creator = FileCreator()
+        test_file_creator.append_file(
+            "index.js", "exports handler = (event, context, callback) => {callback(null, event);}"
+        )
+
+        dirname = test_file_creator.rootdir
+
+        expected_files = {"index.js"}
+
+        random_name = "".join(random.choice(string.ascii_letters) for _ in range(10))
+        outfile = os.path.join(tempfile.gettempdir(), random_name)
+
+        zipfile_name = None
+        try:
+            zipfile_name = make_zip_with_lambda_permissions(outfile, dirname)
+
+            test_zip_file = zipfile.ZipFile(zipfile_name, "r")
+            with closing(test_zip_file) as zf:
+                files_in_zip = set()
+                external_attr_mask = 65535 << 16
+                for info in zf.infolist():
+                    files_in_zip.add(info.filename)
+                    permission_bits = (info.external_attr & external_attr_mask) >> 16
+                    if not platform.system().lower() == "windows":
+                        if info.is_dir():
+                            self.assertEqual(permission_bits, 0o100755)
+                        else:
+                            self.assertEqual(permission_bits, 0o100644)
+                    else:
+                        self.assertEqual(permission_bits, 0o100755)
```

Comment:
I might be miss reading this but looks like if you are not on windows, all permissions are 755? Shouldn't there be a directory vs file check here too?

Reply:
I have kept the permissions as-is because of the previous change increased the permissions overall, it did not matter if it was a directory or a file.